indico/modules/rb_new/client/js/common/bookings/BookingDetailsModal.jsx

```diff
@@ -0,0 +1,263 @@
+                </Message.Header>
+                {reason}
+            </Message.Content>
+        </Message>
+    );
+
+    renderActionButtons = (canAccept, canCancel, canReject, isAccepted) => (
+        <div styleName="action-buttons">
+            {canCancel && (
+                <Button type="button"
+                        icon="remove circle"
+                        size="small"
+                        content={<Translate>Cancel booking</Translate>} />
+            )}
+            {canReject && (
+                <Button type="button"
+                        icon="remove circle"
+                        color="red"
+                        size="small"
+                        content={<Translate>Reject booking</Translate>} />
+            )}
+            {!isAccepted && canAccept && (
+                <Button type="button"
+                        icon="check circle"
+                        color="green"
+                        size="small"
+                        content={<Translate>Accept booking</Translate>} />
+            )}
+        </div>
+    );
+
+    render() {
+        const {
+            booking: {
+                room, bookedForUser, startDt, endDt, repetition, createdByUser, createdDt, bookingReason,
+                occurrences, dateRange, editLogs, canAccept, canCancel, canDelete, canModify, canReject, isCancelled,
+                isRejected, isAccepted
+            }
+        } = this.props;
+        const {occurrencesVisible} = this.state;
+        const dates = {startDate: startDt, endDate: endDt};
+        const createdOn = serializeDate(toMoment(createdDt));
+        const times = {startTime: moment(startDt).utc().format('HH:mm'), endTime: moment(endDt).utc().format('HH:mm')};
+        const recurrence = this.getRecurrence(repetition);
+        const legendLabels = [
+            {label: Translate.string('Occurrence'), color: 'orange'},
+        ];
+        const showActionButtons = (!isCancelled && !isRejected);
+        return (
+            <Modal open onClose={this.handleCloseModal} size="large" closeIcon>
+                <Modal.Header styleName="booking-modal-header">
+                    <Translate>Booking Details</Translate>
+                    <span>
+                        {canModify && <Button icon="pencil" circular />}
+                        {canDelete && <Button icon="trash" circular />}
+                    </span>
+                </Modal.Header>
+                <Modal.Content>
+                    <Grid columns={2}>
+                        <Grid.Column>
+                            <RoomBasicDetails room={room} />
+                            <RoomKeyLocation room={room} />
+                            <TimeInformation recurrence={recurrence}
+                                             dates={dates}
+                                             timeSlot={times}
+                                             onClickOccurrences={() => this.setState({occurrencesVisible: true})}
```

Comment:
what about adding a `toggleOccurrences` method (or showOccurrences and hideOccurrences), and just doing `onClickOccurrences={this.toggleOccurrences}` or `onClickOccurrences={this.showOccurrences}` here?

It doesn't make a huge difference in this case, but it's always better if defining new functions in render() can be avoided (those functions are redefined on each render after all)